smart-wallets/n-key-wallet/n-key-wallet.repl

```diff
@@ -0,0 +1,241 @@
+;;Load contracts
+(begin-tx)
+(env-data {
+  "admin-keyset": ["wallet-admin"]
+  })
+(env-keys ["wallet-admin"])
+(load "../coin-contract/fungible-v2.pact")
+(load "../coin-contract/coin.pact")
+
+;;Load admin key
+(env-data {
+  "admin-keyset": ["wallet-admin"]
+  })
+
+(load "n-key-wallet.pact")
+(commit-tx)
+
+(begin-tx)
+;;Load Account keysets
+(env-data
+  {"acct1": {
+        "keys": [
+            "acct1-key"
+        ],
+        "pred": "keys-all"
+    },
+  "acct2": {
+          "keys": [
+              "acct2-key"
+          ],
+          "pred": "keys-all"
+      },
+  "new": {
+    "keys": [
+        "new-key"
+    ],
+    "pred": "keys-all"
+    },
+    "guardian-guard": {
+      "keys": [
+          "guardian-key"
+      ],
+      "pred": "keys-all"
+      }
+      })
+
+(use n-key-wallet)
+
+;;Create wallet account, acct1
+(env-keys ["guardian-key"])
+(expect "acct1 is created as a smart user" "Write succeeded"
+  (create-smart-user "acct1" [(read-keyset 'acct1) (read-keyset 'acct2)] (read-keyset 'guardian-guard)))
+(expect "acct1 exists in coin table"
+  {"account": "acct1", "balance": 0.0, "guard": (smart-guard)}
+  (coin.details 'acct1))
+
+;;Fund acct1
+(test-capability (coin.COINBASE))
+(expect "acct1 is funded using coinbase" "Write succeeded"(coin.coinbase "acct1" (smart-guard) 100.0))
+(expect "acct1 balance is 100.0" 100.0
+  (at 'balance (coin.details 'acct1)))
+
+;;Test utility functions
+(commit-tx)
+;;logic
+(use n-key-wallet)
+(expect "50 / 100 when 6 keys are in the db" 3 (logic 6 50.0 100.0))
+(expect "51 / 100 when 6 keys are in the db" 4 (logic 6 51.0 100.0))
+(expect "51 / 100 when 2 keys are in the db" 2 (logic 2 51.0 100.0))
+(expect "1 / 100 when 2 keys are in the db" 1 (logic 2 1.0 100.0))
+
+;;numKeys Needed of acct1 to sign
+(expect "acct1 to sign transfer of 51.0" 2 (numKeysNeeded "acct1" 51.0))
+(expect "acct1 to sign transfer of 1.0" 1 (numKeysNeeded "acct1" 1.0))
+
+;;numKeys signed
+(env-keys [])
+(expect "0 keys are signed" 0 (numKeysSigned "acct1"))
+
+(env-keys ["acct1-key", "noone's-key"])
+(expect "1 keys are signed" 1 (numKeysSigned "acct1"))
+
+(env-keys ["acct2-key", "noone's-key"])
+(expect "1 keys are signed" 1 (numKeysSigned "acct1"))
+
+(env-keys ["acct1-key", "acct2-key", "noone's-key"])
+(expect "2 keys are signed" 2 (numKeysSigned "acct1"))
+
+;;Test smart transfer
+
+;;create a none-smart wallet user
+(begin-tx)
+(coin.create-account "acct2" (read-keyset 'new))
+(commit-tx)
+
+;;Test coin.transfer (will fail unless admin-key is signed)
+(begin-tx)
+(use n-key-wallet)
+
+(env-sigs [
+  ;;Bring TRANSFER capability into scope
+  {'key: "dumb-key", 'caps: [(coin.TRANSFER "acct1" "acct2" 100.0)]},
```

Comment:
This isn't needed with direct install as noted above in comment on https://github.com/kadena-io/pact-examples/pull/5/files#diff-5b7e03d26105d205965567929d6f1b19R39